resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -184,4 +183,18 @@ object YarnSparkHadoopUtil {
     ConverterUtils.toContainerId(containerIdString)
   }
 
+  /**
+   * Convert MEMORY_OFFHEAP_SIZE to MB Unit, return 0 if MEMORY_OFFHEAP_ENABLED is false.
+   */
+  def executorOffHeapMemorySizeAsMb(sparkConf: SparkConf): Int = {
+    if (sparkConf.get(MEMORY_OFFHEAP_ENABLED)) {
+      val sizeInMB =
+        sparkConf.getSizeAsMb(MEMORY_OFFHEAP_SIZE.key, MEMORY_OFFHEAP_SIZE.defaultValueString).toInt
+      require(sizeInMB > 0,
+        s"${MEMORY_OFFHEAP_SIZE.key} must be > 0 when ${MEMORY_OFFHEAP_ENABLED.key} == true")
```

Comment:
Please check if `MEMORY_OFFHEAP_SIZE` could equal to 0. The definition of `MEMORY_OFFHEAP_SIZE` checks that it could be `>= 0`.

Reply:
A little conflict with the `MemoryManager` as following:
![image](https://user-images.githubusercontent.com/1475305/62699935-3225e500-ba13-11e9-858e-95da11b2e1ce.png)

if `MEMORY_OFFHEAP_ENABLED` is enable, `MemoryManager.tungstenMemoryMode` will enter `OFF_HEAP` branch and need `MEMORY_OFFHEAP_SIZE  > 0`

Reply:
Then I think we should change the code here.

https://github.com/apache/spark/blob/1941d35d1e250f707c0d70782e568c0887741dfa/core/src/main/scala/org/apache/spark/internal/config/package.scala#L234

Reply:
0 is defaultValue,  change to 
```
.checkValue(_ > 0, "The off-heap memory size must be positive")
    .createWithDefault(1)
```
?

Reply:
Maybe we should give a suitable defaultValue ,like 1073741824(1g)?

Reply:
@jerryshao Do we need to change 0 to a  suitable defaultValue?

Reply:
I see, then I would suggest not to change it. Seems there's no good value which could cover most of the scenarios, so good to leave as it is.

Reply:
ok~

Reply:
its odd that check is >= 0 in the config, seems like we should change but can you file a separate jira for that?

Reply:
OK~ I will add a new JIRA to discuss this issue.